src/clr-angular/utils/smart-popover/smart-anchor.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ *  Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ *
+ */
+
+import { Directive, Input, ElementRef } from '@angular/core';
+import { ClrPosition } from '../enums/position.enum';
+import { ClrSmartPopoverEventsService } from './providers/smart-popover-events.service';
+import { ClrSmartPopoverPositionService } from './providers/smart-popover-position.service';
+import { ClrAlignment } from './enums/alignment.enum';
+
+@Directive({
+  selector: '[clrSmartAnchor]',
+  host: {
+    '[class.clr-anchor]': 'true',
+  },
+})
+export class ClrSmartPopoverAnchor {
+  constructor(private smartEventService: ClrSmartPopoverEventsService, private element: ElementRef) {
```

Comment:
Neither of these arguments need to be private, since you are only using it in the constructor.

Reply:
Will remove